src/res/KM_ResKeys.pas

```diff
@@ -102,20 +132,36 @@ procedure TKMKeyLibrary.LoadKeymapFile;
 
     if not InRange(keyId, 0, fCount - 1) or (keyVal = -1) then Continue;
 
-    fKeys[keyId] := keyVal;
+    fFuncs[keyId].Key := keyVal;
+    fFuncs[keyId].TextId := KEY_FUNC_TX[keyId];
+
+    if (keyId in [0..3]) or (keyId in [24..26]) or (keyId in [40..44]) then
```

Comment:
this could be rewritten into a more compact "case keyId of 0..3, 24,26,40..44: ..."